core/src/main/java/org/infinispan/stats/impl/StatsImpl.java

```diff
@@ -103,11 +127,15 @@ public StatsImpl(Stats other) {
          statsMap.put(AVERAGE_WRITE_TIME_NANOS, other.getAverageWriteTimeNanos());
          statsMap.put(REQUIRED_MIN_NODES, (long) other.getRequiredMinimumNumberOfNodes());
       } else {
-         for (String key : Attributes)
-            statsMap.put(key, -1L);
+         emptyStats();
       }
    }
 
+   private void emptyStats() {
+      for (String key : Attributes)
```

Comment:
I'm terribly shocked seeing Attributes is not final :)

Reply:
ok